src/vos/tests/vts_mvcc.c

```diff
@@ -705,9 +951,20 @@ static struct op operations[] = {
 	{"listd",	T_R,	L_D,	L_NIL,	R_R,	W_NIL,	listd_f},
 	{"lista",	T_R,	L_A,	L_NIL,	R_R,	W_NIL,	lista_f},
 	{"listr",	T_R,	L_A,	L_NIL,	R_R,	W_NIL,	listr_f},
-	{"queryc",	T_R,	L_C,	L_NIL,	R_R,	W_NIL,	NULL},
-	{"queryo",	T_R,	L_O,	L_NIL,	R_R,	W_NIL,	NULL},
-	{"queryd",	T_R,	L_D,	L_NIL,	R_R,	W_NIL,	NULL},
+	{"querymaxd",	T_R,	L_D,	L_NIL,	R_NE,	W_NIL,	querymax_d},
```

Comment:
For queries' rtypes, R_NE seems reasonable if queries return -DER_NONEXISTs when the rlevel entities is empty. @jolivier23, @mchaarawi, is that the case?

Reply:
Thanks for your feedback @liw , I just sent a patch with the changes for rlevels, let me know if I missed something

Reply:
@liw The issue I ran into on some of these, for example maxdr, is that it needs an akey and will dereference L_O[3] which is outside the bounds so I got back wrong results when it was initially set higher for such cases.   I think the level can safely be the highest requested level but for dr, it needs to be L_A because it will actually use the akey.

Reply:
Indeed... "dr" cases are special. Could we set their rlevel to L_O and replace their set_akey calls with code that set akeys arbitrarily (for we don't care what their akeys are) without reading any paths?

Reply:
It's a little tricky since vos_query_key returns -DER_NONEXIST if it can't find anything.  I think the mvcc test would expect it to find something ?   That said, it would probably be good to test both cases for coverage (e.g. both with a known and missing dkey)

Reply:
Hi guys, I sent a patch using rlevel L_O for dr cases and it is working fine. 
@liw what do you mean by set_akey without reading any paths?

Reply:
@liw Do you have objections to landing the patch as is and figuring out how to work this later?